blog/17-05-2023-using-celo's-identity-protocol-to-create-a-decentralized-kyc-solution/using-celo's-identity-protocol-to-create-a-decentralized-kyc-solution.md

```diff
@@ -0,0 +1,280 @@
+Ensure you run the command above in the root directory of your project.
+
+Open your project in Visual Studio code or your preferred IDE. It should have the following directories/files in it:
+
+![Directory Output](./images//directory_output.png)
+
+### Creating the Smart Contract
+1. First, let's create a new file named KYC.sol in the contracts directory and write the smart contract code to store and verify user data:
+
+```solidity
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+contract KYC {
+    struct UserData {
+        string name;
+        string documentId;
+        bool isVerified;
+    }
+
+    mapping(address => UserData) public userData;
+    address public admin;
+
+    constructor() {
+        admin = msg.sender;
+    }
+
+    modifier onlyAdmin() {
+        require(msg.sender == admin, "Only admin can perform this action");
+        _;
+    }
+
+    function addUser(address _user, string memory _name, string memory _documentId) public onlyAdmin {
+        userData[_user] = UserData(_name, _documentId, false);
+    }
+
+    function verifyUser(address _user) public onlyAdmin {
+        userData[_user].isVerified = true;
+    }
+
+    function isUserVerified(address _user) public view returns (bool) {
+        return userData[_user].isVerified;
+    }
+}
+ ```
+
+This is a Solidity smart contract that represents the core of the decentralized KYC system.
+- A `UserData` struct is defined to store user information, including their name, document ID, and whether they are verified.
+- A mapping is created to associate user addresses with their `UserData`.
+- An admin address variable is used to store the address of the contract creator.
+- The constructor sets the admin to the address that deploys the contract.
+- A modifier `onlyAdmin` is used to restrict certain functions to be callable only by the admin.
+- The `addUser` function allows the admin to add a user with their name and document ID. It sets the `isVerified` flag to false by default.
+- The `verifyUser` function enables the admin to mark a user as verified.
+- The `isUserVerified` function returns the isVerified status of a user.
+
+2. Create a file named `2_deploy_contracts.js` in the migrations folder and populate it with the code below.
+
+```js
+var KYC = artifacts.require("KYC");
+
+module.exports = function (deployer) {
+  deployer.deploy(KYC);
+};
+ ```
+
+The code defines a deployment script for the KYC contract, using the Truffle framework's deployer function. The script specifies that the KYC contract will be deployed, and then calls the deploy function with KYC as its argument to deploy the contract.
+
+### Deploying the Smart Contract
+
+To deploy the smart contract, we need to compile it first. In your terminal, run the following command in the root directory of the project:
+
+```bash
+truffle compile
+```
+
+This will compile the smart contract and generate the artifacts in the /build/contracts directory. You should get an output similar to this:
```

Comment:
```suggestion
This will compile the smart contract and generate the artifacts in the `/build/contracts` directory. You should get an output similar to this:
```